pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertEqual(
+                    task_body['progress_report']['puppet_importer']['metadata']['error_message'],  # noqa pylint:disable=line-too-long
+                    None
+                )
+
+
+class SyncInvalidFeedTestCase(_BaseTestCase):
+    """If an invalid feed is given, the sync completes with reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create puppet repository with an invalid feed and sync it."""
+        super(SyncInvalidFeedTestCase, cls).setUpClass()
+        body = _gen_puppet_repo_body()
+        body['importer_config']['feed'] = ''.join(['http://', uuid4()])
+        cls.attrs_iter = (create_repository(cls.cfg, body),)
+        cls.sync_repo = []  # raw responses
+        cls.task_bodies = tuple((chain.from_iterable(  # response bodies
+            poll_spawned_tasks(cls.cfg, call_report)
+            for call_report in (_sync_repo(
+                cls.cfg,
+                attr_iter['_href'],
+                cls.sync_repo
+            ) for attr_iter in cls.attrs_iter)
+        )))
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        self.assertEqual(self.sync_repo[0].status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['error'])
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['traceback'])
+
+    def test_error_details(self):
+        """Assert each task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertNotEqual(
+                    task_body['progress_report']['puppet_importer']['metadata']['error'],  # noqa pylint:disable=line-too-long
+                    None
+                )
+
+    def test_number_tasks(self):
+        """Assert that only one task was spawned."""
+        self.assertEqual(len(self.task_bodies), 1)
+
+
+class PublishTestCase(_BaseTestCase):
+    """Upload custom puppet module to a repository, publish and download it."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Upload puppet module to a repo, copy it to another, publish and download.
+
+        Create two puppet repositories, both without feeds. Upload an module to
+        the first repository. Copy this content to the second repository. Add a
+        distributor to the first repository, publish it, and download
+        the module uploaded earlier.
```

Comment:
The contents of this test differ from what is described here. Puppet modules are not downloaded to one repository and then copied to the second. Is there a reason for the difference?